### PR TITLE
[fix](function) array distance should return null when result is nan

### PR DESCRIPTION
**Pull request by morrySnow**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

File: be/src/vec/functions/array/function_array_distance.h [hunk 1]
```diff
@@ -160,7 +160,11 @@ class FunctionArrayDistance : public IFunction {
                                          nested_col2->get_float64(pos));
             }
             if (!dst_null_data[row]) {
-                dst_data[row] = DistanceImpl::finalize(st);
+                if (double final_value = DistanceImpl::finalize(st); std::isnan(final_value)) {
```
Comment (HappenLee) [inline be/src/vec/functions/array/function_array_distance.h:163]: dst_null_data[row] = std::isnan(dst_data[row]);

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by morrySnow** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**